src/data_structures/array_2d.rs

```diff
@@ -0,0 +1,55 @@
+use std::fmt;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct Array2D<T> {
+    array: Vec<T>,
+    width: usize,
+    height: usize,
+}
+
+impl<T> Array2D<T> {
+    pub fn new(width: usize, height: usize) -> Self {
+        Array2D {
+            array: Vec::with_capacity(width * height),
```

Comment:
Since we seem to know the size already at construction time, can we implement the `Sized` trait? Is this desirable or efficient?

Reply:
No, because it's internally using a Vec, which isn't `Sized`.